src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -4219,7 +4219,14 @@ LogicalResult ONNXScatterElementsOp::inferShapes(
 
 LogicalResult ONNXScatterNDOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  if (!data().getType().isa<mlir::RankedTensorType>() ||
+      !indices().getType().isa<mlir::RankedTensorType>() ||
+      !updates().getType().isa<mlir::RankedTensorType>()) {
+    return failure();
```

Comment:
We always return `success()` instead of `failure()` here since we may run shape inference multiple times.

Reply:
OK, got it